specs/withdrawals.md

```diff
@@ -0,0 +1,176 @@
+# Withdrawals
+
+<!-- All glossary references in this file. -->
+[g-deposited]: glossary.md#deposited-transaction
+[g-withdrawal]: glossary.md#withdrawal
+[g-mpt]: glossary.md#merkle-patricia-trie
+[g-relayer]: glossary.md#withdrawals
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Constants](#constants)
+- [Data structures](#data-structures)
+- [Withdrawal verification](#withdrawal-verification)
+- [Withdrawal Flow](#withdrawal-flow)
+  - [Rationale](#rationale)
+- [The L2 Withdrawal Contract](#the-l2-withdrawal-contract)
+- [Trie Interface](#trie-interface)
+- [Key differences from Current System](#key-differences-from-current-system)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Withdrawals are transactions initiated on L2, and executed on L1. They may be used to transfer data, ether or both from
+L1 to L2.
+
+**Vocabulary note**:
+
+- _withdrawing transaction_ refers specifically to an L2 transaction
+- _withdrawn transaction_ refers specifically to an L1 transaction
+- _withdrawal_ can refer to the transaction at various stages
+
+## Constants
+
+| Name                  | Value     | Unit    |
+| --------------------- | --------- | ------- |
+| `FINALIZATION_WINDOW` | `604_800` | seconds |
+
+This `FINALIZATION_WINDOW` value is equivalent to 7 days.
+
+## Data Structures and Type Aliases
+
+1. A `WithdrawalMessage` is encoded in a struct as follows:
+
+    ```js
+    struct WithdrawalMessage {
+        uint256 nonce;
+        address sender;
+        uint256 value;
+        bytes message;
+    }
+    ```
+
+1. The `L2OutputTimestamp` is an alias for `uint256`, and MUST be a multiple of the `SUBMISSION_INTERVAL` described
+  in the [L2 Output](./proposals.md#constants) document.
+
+1. The `WithdrawalsRoot` is an alias for a `bytes32` value, corresponding to the [MPT][g-mpt]
+  storage root of the Withdrawals predeploy contract at `0x4200000000000000000000000000000000000015` (described below).
+
+1. The `WithdrawalsRootInclusionProof` proof contains the data necessary to prove that the provided `WithdrawalsRoot` is
+  included in the [SSZ merkleization](https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#merkleization)
+  of the `L2Output` as defined in the
+  [L2 output commitment construction](./proposals.md#l2-output-commitment-construction).
+
+1. A `WithdrawalMessageInclusionProof` is an MPT proof encoded in a struct as follows:
+
+   ```js
+   struct WithdrawalMessageInclusionProof {
+       WithdrawalMessage message;
+       bytes32 l2withdrawalsRoot;
+       bytes memory _key;      // storage key of the withdrawal message commitment
+       bytes memory _value;    // Always bytes32(1) (boolean validity status of the message)
```

Comment:
Why does this have to be included in the struct if it is a constant?

Reply:
Good point, it does not.